### PR TITLE
feat(pose_initializer): add a method type to the `/localization/initialize` service.

### PR DESCRIPTION
**Pull request by YamatoAndo**
## Description

I have added the capability to directly initialize by the input pose of the `/localization/initialize` service.
To facilitate this, a new service message is being defined https://github.com/tier4/tier4_autoware_msgs/pull/121, and I have changed service message type of `/localization/initialize`
`autoware_adapi_v1_msgs::srv::InitializeLocalization` -> `tier4_localization_msgs::srv::InitializeLocalization`.
Along with that I have modified following packages
- pose_initilizer
- component_interface_specs
- default_ad_api

## Tests performed

- test case1
```
ros2 service call /localization/initialize tier4_localization_msgs/srv/InitializeLocalization
```
The initial position is automatically estimated using the GNSS pose.

- test case2
```
ros2 service call /localization/initialize tier4_localization_msgs/srv/InitializeLocalization "
method: 0
"
```
The initial position is automatically estimated using the GNSS pose.

- test case3
```
ros2 service call /localization/initialize tier4_localization_msgs/srv/InitializeLocalization "
pose_with_covariance:
  - header:
      frame_id: map
    pose:
      pose:
        position:
          x: 89571.1640625
          y: 42301.1875
          z: -3.1565165519714355
        orientation:
          x: 0.0
          y: 0.0
          z: 0.28072773940524687
          w: 0.9597874433062874
      covariance: [0.25, 0, 0, 0, 0, 0, 0, 0.25, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.06853891909122467]
method: 0
"
```
The initial position is automatically estimated using the input pose.

- test case4

```
ros2 service call /localization/initialize tier4_localization_msgs/srv/InitializeLocalization "
pose_with_covariance:
  - header:
      frame_id: map
    pose:
      pose:
        position:
          x: 89571.1640625
          y: 42301.1875
          z: -3.1565165519714355
        orientation:
          x: 0.0
          y: 0.0
          z: 0.28072773940524687
          w: 0.9597874433062874
      covariance: [0.25, 0, 0, 0, 0, 0, 0, 0.25, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.06853891909122467]
method: 1
"
```
The initial position is set directly by the input pose.

## Notes for reviewers

This PR should be merged after https://github.com/tier4/tier4_autoware_msgs/pull/121.
Otherwise, the build will fail, and the GitHub CI will not pass.

And this PR should be merged with https://github.com/tier4/tier4_ad_api_adaptor/pull/112 at the same time.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: localization/pose_initializer/src/pose_initializer/pose_initializer_core.cpp [hunk 1]
```diff
@@ -147,25 +149,52 @@ void PoseInitializer::on_initialize(
...
+      pub_reset_->publish(pose);
+
+      change_node_trigger(true, false);
+      res->status.success = true;
+      change_state(State::Message::INITIALIZED);
+
+    } else if (req->method == Initialize::Service::Request::DIRECT) {
+      if (req->pose_with_covariance.empty()) {
+        std::stringstream message;
+        message << "No input pose_with_covariance. If you wanna use DIRECT method, please input "
```
Comment (mitsudome-r) [inline localization/pose_initializer/src/pose_initializer/pose_initializer_core.cpp:175]: I think it's better to keep expression more formal for error messages.

```suggestion
        message << "No input pose_with_covariance. If you want to use DIRECT method, please input "
```

**Review by isamu-takagi — APPROVED** [review]
LGTM for default_ad_api package

**Review by yukkysaito — APPROVED** [review]
LGTM

**Review by KYabuuchi — COMMENTED** [review]
I think the feature is fine. :clap:

Since the method of setting the initial position via CLI is useful, how about writing it in the README of pose_initializer?
For example, a chapter like `Direct initial position set via CLI`.

**Comment by SakodaShintaro** [comment]
Thank you.
In the explanation, it looked like it would be fine to just merge the PRs in autoware.universe and tier4_autoware_msgs, but I got a build error.
Is it correct that [the PR that Takagi-san linked to](https://github.com/tier4/tier4_ad_api_adaptor/pull/112) also needs to be merged at the same time?

**Comment by YamatoAndo** [comment]
@SakodaShintaro 
> Is it correct that https://github.com/tier4/tier4_ad_api_adaptor/pull/112 also needs to be merged at the same time?
Yes, I'm sorry.

**Comment by YamatoAndo** [comment]
@KYabuuchi I added some descriptions to initialize pose the via CLI

**Review by SakodaShintaro — APPROVED** [review]
Looks Good To Me

**Pull request merged**